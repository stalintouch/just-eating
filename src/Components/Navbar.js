import React, { Component } from 'react';
import AppBar from 'material-ui/AppBar';
import IconButton from 'material-ui/IconButton';
import FlatButton from 'material-ui/FlatButton';
import focus from '../scripts/focus'

export default class AppBarExampleIconButton extends Component {

  // this function auto focus on the search bar when the navbar is clicked
  handleClick = () => {
    focus('autoComplete');
  };

  render() {
    const styles = {
      title: {
        cursor: 'pointer'
      }
    };

    return (
      <AppBar
        title={<span style={styles.title}>Just Eat it!</span>}
        onTitleClick={this.handleClick}
        onRightIconButtonClick={this.handleClick}
        iconElementLeft={<IconButton />}
        iconElementRight={<FlatButton label="Find Restaurants nearby" />}
      />
    );
  }
}
