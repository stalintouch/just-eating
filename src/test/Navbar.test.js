import Navbar from '../Components/Navbar';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

describe('<Navbar />', () => {
  let wrapperShallow;
  let wrapperMount;

  beforeEach(() => {
    wrapperShallow = shallow(<Navbar />);

    wrapperMount = mount(
      <MuiThemeProvider>
        <Navbar />
      </MuiThemeProvider>
    );
  });

  it('renders correctly', () => {
    expect(wrapperShallow).toMatchSnapshot();
  });

  it('selects the navbar text', () => {
    const text = wrapperMount
      .find('span')
      .first()
      .text();
    expect(text).toEqual('Just Eat it!');
  });

  it('focus on search bar when clicked', () => {
    const leftSpan = wrapperMount.find('span').first();
    leftSpan.simulate('click', 'autoComplete');
  });
});
