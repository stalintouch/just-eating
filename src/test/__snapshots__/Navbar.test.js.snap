// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AppBarExampleIconButton />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "iconElementLeft": <IconButton
        disableTouchRipple={false}
        disabled={false}
        iconStyle={Object {}}
        tooltipPosition="bottom-center"
        touch={false}
/>,
      "iconElementRight": <FlatButton
        disabled={false}
        fullWidth={false}
        label="Find Restaurants nearby"
        labelPosition="after"
        labelStyle={Object {}}
        onKeyboardFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        primary={false}
        secondary={false}
/>,
      "onRightIconButtonClick": [Function],
      "onTitleClick": [Function],
      "showMenuIconButton": true,
      "title": <span
        style={
                Object {
                        "cursor": "pointer",
                      }
        }
>
        Just Eat it!
</span>,
      "zDepth": 1,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "iconElementLeft": <IconButton
          disableTouchRipple={false}
          disabled={false}
          iconStyle={Object {}}
          tooltipPosition="bottom-center"
          touch={false}
/>,
        "iconElementRight": <FlatButton
          disabled={false}
          fullWidth={false}
          label="Find Restaurants nearby"
          labelPosition="after"
          labelStyle={Object {}}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          primary={false}
          secondary={false}
/>,
        "onRightIconButtonClick": [Function],
        "onTitleClick": [Function],
        "showMenuIconButton": true,
        "title": <span
          style={
                    Object {
                              "cursor": "pointer",
                            }
          }
>
          Just Eat it!
</span>,
        "zDepth": 1,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
